File: solution/00/05/sol_brute.py
def is_palindrome(s: str) -> str:
    return s == s[::-1]


class Solution:
    def longestPalindrome(self, s: str) -> str:  # noqa
        max_len = 1
        p = s[0]
        n = len(s)
        for i in range(n):
            for j in range(i + 1, n + 1):
                if is_palindrome(s[i:j]):
                    if len(s[i:j]) > max_len:
                        max_len = len(s[i:j])
                        p = s[i:j]

        return p


def main():
    s = Solution()
    assert (result := s.longestPalindrome("babad")) == "bab", f"unexpected {result=}"
    assert (result := s.longestPalindrome("cbbd")) == "bb", f"unexpected {result=}"
    assert (result := s.longestPalindrome("a")) == "a", f"unexpected {result=}"

    print("PASSED")


if __name__ == '__main__':
    main()

File: solution/00/05/sol_brute.cc
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    static string longestPalindrome(const string& s)
    {
        const auto n = s.size();
        if (n == 0) return "";

        int max_len = 1;
        int start = 0;

        auto isPalindrome = [&](int left, int right)
        {
            while (left < right)
            {
                if (s[left] != s[right]) return false;
                ++left;
                --right;
            }
            return true;
        };

        for (int i = 0; i < n; ++i)
        {
            for (int j = i; j < n; ++j)
            {
                if (isPalindrome(i, j) && j - i + 1 > max_len)
                {
                    start = i;
                    max_len = j - i + 1;
                }
            }
        }

        return s.substr(start, max_len);
    }
};

int main()
{
    const string s = "babad";
    cout << Solution::longestPalindrome(s) << endl;
    return 0;
}
